### DISCUSSION TITLE
Mouse events under WSL(2)

### DISCUSSION THREAD

**Discussion by danmur** [kingToolbox/WindTerm#275] [Q&A]
Hello! I've noticed mouse events don't work under WSL2 (e.g. clicking headers in htop). I'm guessing it's a more complicated thing, like using a different interface instead of the wsl command (mobaxterm also has the same problem if you use the wsl command, but not the 'native connection', whatever that is) but I just wanted to check if there were any settings I could use to make it work?

Thanks, and love your work, I use it full time for my day job :)

**Comment by kingToolbox** [comment]
I tested `WSL.exe` and found that it supports `VT mouse`. I'm not sure what went wrong, because `WindTerm` supports the `VT mouse` very well.

I will investigate this issue, you can temporarily connect to `Linux distributions` in `WSL` via `SSH`, so that you can use the mouse to operate `htop`. I will update here if there is any progress, thank you.

**Reply by danmur** [reply]
Thank you!

**Reply by kingToolbox** [reply]
I have been studying how to make `WindTerm` support the mouse in `WSL` for the past few days, but unfortunately, because `Microsoft` hardly provided relevant development documents, even after reading a lot of code and issues of `Windows Terminal`, I still have no success.

I will continue to explore, but I need to start making the `MacOS` universal version from next week, so the progress may slow down, please understand. If there is any progress, I will still update here. Please stay tuned, thank you.

**Reply by danmur** [reply]
Thanks, appreciate it :). It's not urgent for me was just surprising as it worked elsewhere.

Yes, I did a bit of reading and couldn't find much at all. I'll just paste the screenshot of mobaxterm here since it was the closest hint I saw:
![image](https://user-images.githubusercontent.com/613894/143354995-b076b5b8-3e04-4cbd-91f7-6833dd096d78.png)

"wsl command" did not work in mobaxterm but "native connector" does (I have no idea what the native connector is)

**Reply by kingToolbox** [reply]
I have not been able to find relevant documentation. But once I solve this problem, I will tell you how `WSL` can support the mouse. I guess it might just set a certain parameter or environment variable when running `WSL`, it shouldn't be too complicated.
